Спринт 16

ValentinPside opened the pull request:
1я попытка

ValentinPside committed:
Спринт 16

1я попытка

faritowich commented on app/src/main/java/com/example/playlistmaker/search/creator/Creator.kt:
```diff
@@ -0,0 +1,19 @@
+
+    fun createiTunesService() : ITunesSearchAPI {
+        return createRetrofit().create(ITunesSearchAPI::class.java)
+    }
```
Метод нигде не используется. Вообще я бы точно так же в Creator создавал бы SearchTracksRepository и другие сущности

faritowich commented on app/src/main/java/com/example/playlistmaker/player/ui/AudioPlayerActivity.kt:
```diff
@@ -10,12 +9,19 @@ import android.view.View
+    private lateinit var playerViewModel: AudioPlayerViewModel
 
     companion object {
         private const val STATE_DEFAULT = 0
```
Вместо этих констант лучше создать enum class для описания состояния

faritowich commented on app/src/main/java/com/example/playlistmaker/settings/ui/SettingsViewModel.kt:
```diff
@@ -0,0 +1,56 @@
+    }
+}
+
+data class ViewState(
```
тут и в других ViewState - private

faritowich commented on app/src/main/java/com/example/playlistmaker/search/ui/SearchViewModel.kt:
```diff
@@ -0,0 +1,76 @@
+        viewState.value = viewState.value?.copy(placeHolderState = PlaceHolderState.LOADING)
+    }
+
+    fun showCommunicationProblemPlaceholder(){
```
private

ValentinPside committed:
Спринт 16

2я попытка

ValentinPside commented:
2я попытка

ValentinPside committed:
Спринт 16 (3я попытка)

Добавил implementation 'com.squareup.okhttp3:okhttp:4.11.0'
Проверил на API33, всё заработало

ValentinPside commented:
Добавил implementation 'com.squareup.okhttp3:okhttp:4.11.0'
Проверил на API33, всё заработало
![2023-10-30_23-37-28](https://github.com/ValentinPside/Playlist-Maker/assets/91324342/66e90966-5b99-47f0-9d6e-217a935f45dd)
![2023-10-30_23-36-46](https://github.com/ValentinPside/Playlist-Maker/assets/91324342/ccc4de3b-4e31-43f0-a446-ef0c340975d4)
